backend/src/main/java/com/bakdata/conquery/util/search/solr/FilterValueSearch.java

```diff
@@ -0,0 +1,267 @@
+package com.bakdata.conquery.util.search.solr;
+
+import java.io.IOException;
+import java.net.URLDecoder;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import javax.annotation.CheckForNull;
+import javax.annotation.Nullable;
+
+import com.bakdata.conquery.apiv1.frontend.FrontendValue;
+import com.bakdata.conquery.models.config.search.solr.FilterValueConfig;
+import com.bakdata.conquery.models.datasets.concepts.Searchable;
+import com.bakdata.conquery.models.datasets.concepts.filters.Filter;
+import com.bakdata.conquery.models.datasets.concepts.filters.specific.SelectFilter;
+import com.bakdata.conquery.resources.api.ConceptsProcessor;
+import com.bakdata.conquery.resources.api.ConceptsProcessor.AutoCompleteResult;
+import com.bakdata.conquery.util.search.solr.entities.SolrFrontendValue;
+import com.google.common.collect.Iterables;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.text.StringSubstitutor;
+import org.apache.solr.client.solrj.SolrClient;
+import org.apache.solr.client.solrj.SolrQuery;
+import org.apache.solr.client.solrj.SolrServerException;
+import org.apache.solr.client.solrj.StreamingResponseCallback;
+import org.apache.solr.client.solrj.response.QueryResponse;
+import org.apache.solr.client.solrj.util.ClientUtils;
+import org.apache.solr.common.SolrDocument;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Helper class to abstract/capsule the {@link Searchable}s of the filter away.
+ * The data was/is imported to Solr by a {@link FilterValueIndexer} for each {@link Searchable}, because a single {@link Searchable} might be used by multiple {@link Filter}s.
+ */
+@AllArgsConstructor
+@Slf4j
+public class FilterValueSearch {
+
+    private final SelectFilter<?> filter;
+    private final SolrProcessor processor;
+    private final SolrClient solrClient;
+    private final FilterValueConfig filterValueConfig;
+
+    public List<FilterValueIndexer> getSearchesFor(SelectFilter<?> searchable, boolean withEmptySource) {
+        List<Searchable> searchReferences = searchable.getSearchReferences();
```

Comment:
```suggestion
        List<Searchable> searchReferences = new ArrayList(searchable.getSearchReferences());
```

Du verlässt dich hier auf die Implementierung